packages/chevrotain/src/parse/parser/traits/recognizer_engine.ts

```diff
@@ -210,35 +213,30 @@ export class RecognizerEngine {
     this.shortRuleNameToFull[shortName] = ruleName
     this.fullRuleNameToShort[ruleName] = shortName
 
-    function invokeRuleWithTry(this: MixedInParser, args: any[]) {
+    function invokeRuleWithTry(this: MixedInParser, ...args: ARGS[]): R {
       try {
+        this.ruleInvocationStateUpdate(shortName, ruleName, this.subruleIdx)
```

Comment:
now that you simplified this logic, we may be able to to get rid of the conditional related to `outputCst`
- But **not in this PR** its getting large anyhow